### ISSUE TITLE
Set window title

### ISSUE THREAD

**Issue by alexfedoseev** [zeit/hyper#516]
Is it possible to set window title? Not tab's title, but window wrapper's title. Or sync it w/ active tab's title (this seems the right way).

I can set tab title w/ `echo -en "\e]0;My Title\a"`, but window title always stays the same: `HyperTerm - HyperTerm`.

**Comment by albinekb** [comment]
This is on linux right?

This should do it http://electron.atom.io/docs/api/app/#appsetnamename
Do you want me to write a plugin for you?

**Comment by alexfedoseev** [comment]
@albinekb Thanks for help! Comments down below.

> This is on linux right?

Nope, it's MacOS.

> This should do it http://electron.atom.io/docs/api/app/#appsetnamename

Nope, this doesn't work. After this title stays the same:

<details>
 <summary>Screen (title is in the top left corner)</summary>
<img width="1280" alt="screen shot 2016-07-31 at 20 12 08" src="https://cloud.githubusercontent.com/assets/4244251/17276410/647283c4-575b-11e6-9a34-55db4ff3170b.png">
</details>

>Do you want me to write a plugin for you?

Actually it'd be better to clarify a few things, that I can't get about how things work, thus I can do it by myself. AFAIU `app` (which is Electron app) is available only in `onApp` handler. But I don't have access to it in another methods (eg `decorateTerm` or `middleware` etc). I guess what I need is [`remote`](http://electron.atom.io/docs/api/remote/), thus I can communicate w/ main process from the rendered UI. But no idea how to arrange this (if it's possible) in context of Hyperterm api + not sure that I got all things right.

**Comment by alexfedoseev** [comment]
Also I've found this method: http://electron.atom.io/docs/api/browser-window/#winsettitletitle
But when I trigger it in `onWindow` handler, nothing happens (actually the same w/ any other of these methods, eg `minimize`).

Also also I found `window.rpc`, which can `emit` messages to _some_ `window`, but I don't get what this `window` is (instance of `BrowserWindow`?) and where (& how) can I add listener for these messages (in `onWindow`?).

**Comment by alexfedoseev** [comment]
Forgot to mention, that in `app.createWindow` callback it works:

```js
exports.onApp = (app) => {
  app.createWindow(win => win.setTitle('Boo!')); // works!
}
```

**Comment by kbrock** [comment]
For Mac Terminal, the escape sequence you used does not work for me.

The following variations work for me:

```bash
echo -e "\033];My Title\007"
echo -e "\033]0;My Title\007"
echo -e "\033];My Title\a"
echo -e "\e]0;My Title\a" # doesn't work
```

**Comment by leo** [comment]
^

**Comment by alexfedoseev** [comment]
@leo Just for the record: this wasn't resolved.

**Comment by leo** [comment]
@alexfedoseev What's with the solution you've mentioned above?

**Comment by alexfedoseev** [comment]
> What's with the solution you've mentioned above?

AFAIR it works, but only in `exports.onApp`, after this callback is triggered, I have no way to communicate w/ this Electron `app` instance via other handlers, at least I don't know how.

Actually this issue was a part of [bigger one](https://github.com/zeit/hyper/issues/612), where I tried to scaffold predefined layouts, but gave up, since there is no available API for such customizations.

Currently I switched back ti iTerm2, so this issue can be closed, but answers (or comments) on the questions above might be useful for somebody else.
